### DISCUSSION TITLE
Applying RPU_config after loading a HWA-trained model (exmaple 24)

### DISCUSSION THREAD

**Discussion by kusionlion** [IBM/aihwkit#673] [Q&A] [answered]
Hi, 

I tried applying RPU_config after loading a HWA-trained model (exmaple 24) and found that the RPU_config doesn't seem to be used to represent my weights. The reason for this this doubt is that when I scaled the noise in RPU by 0 to 5000 times, the inference results (F1) stay the same. 

Here is the steps I did (example 24):
**1. performed HWA training and saved the trained model:**
ARGS.train_hwa=True
[code link](https://github.com/IBM/aihwkit/blob/69e524fbc2c5462cbe1a12ee1662879c001c30c0/examples/24_bert_on_squad.py#L85-L91)
PARSER.add_argument(
    "-c",
    "--checkpoint",
    help="File name specifying where to load/save a checkpoint",
    default="./**New_trained_results**.pth",
    type=str,
)

**2. Loaded new weights and changed the programming scale**
ARGS.load=True
[code link](https://github.com/IBM/aihwkit/blob/69e524fbc2c5462cbe1a12ee1662879c001c30c0/examples/24_bert_on_squad.py#L164)
noise_model=PCMLikeNoiseModel(**prog_noise_scale=5000**),

The step 2 generates many F1 over time. I varied the prog_noise_scale, and the F1 seems to be independent to the prog_noise_scale. Am I doing it wrong? 

Thanks in advance.

**Comment by maljoras-sony** [comment, marked as answer]
By default, the rpu_config from the checkpoint is loaded. It is not clear from the above, how you changed the rpu_config. You could use `model.drift_analog_weights(noise_model=the_new_noise_model)` that would apply the new noise model without the need to change the rpu_config of the model.

**Comment by PabloCarmona** [comment]
Hello @kusionlion! The answer from @maljoras-sony help you out? Need any further assistance? Thanks!
